app/javascript/packs/foremMobile.js

```diff
@@ -0,0 +1,114 @@
+/* global Runtime, fetchBaseData */
+
+function initializeNamespaceWhenPageIsReady() {
+  setTimeout(() => {
```

Comment:
The current logic does make multiple attempts if the page takes a long time to load. The `else` calls `initializeNamespaceWhenPageIsReady` which will kick-off the next `setTimeout`. Whenever the data is available it will just avoid starting the next timeout. It's the same logic [used for the RuntimeBanner](https://github.com/forem/forem/blob/main/app/javascript/packs/runtimeBanner.jsx#L11-L25)

I can refactor it to use `setInterval`/`clearInterval` but I agree with what you mention of a Promise based check, because this is something we now do in multiple areas. I'll give it a try and see if I can come up with something that's reusable. The problem I see is that `initializePage.js` (from your link) is in the asset pipeline and couldn't `import` a pack, but at least the webpacker codebase could benefit from this.